framework/test/gtest/BundleTrackerMethodTest.cpp

```diff
@@ -0,0 +1,480 @@
+    << "Creation of BundleTracker failed";
+}
+
+TEST_F(BundleTrackerMethodTest, GetBundlesWorks)
+{
+  BundleTracker<> bundleTracker(context, all_states);
+  std::vector<Bundle> bundles = bundleTracker.GetBundles();
+  EXPECT_EQ(0, bundles.size())
+    << "GetBundles() should return an empty vector before Open()";
+
+  ASSERT_NO_THROW(bundleTracker.Open()) << "BundleTracker failed to start";
+  Bundle bundleMain = cppmicroservices::testing::GetBundle("main", context);
+  Bundle bundleSys =
+    cppmicroservices::testing::GetBundle("system_bundle", context);
+  Bundle bundleA = cppmicroservices::testing::InstallLib(
+    framework.GetBundleContext(), "TestBundleA");
+  bundles = bundleTracker.GetBundles();
+
+  bool mainTracked =
+    std::find(bundles.begin(), bundles.end(), bundleMain) != bundles.end();
+  bool sysTracked =
+    std::find(bundles.begin(), bundles.end(), bundleSys) != bundles.end();
+  bool bundleATracked =
+    std::find(bundles.begin(), bundles.end(), bundleA) != bundles.end();
+
+  EXPECT_TRUE(mainTracked) << "GetBundles() should include the main bundle";
+  EXPECT_TRUE(sysTracked) << "GetBundles() should include the system bundle";
+  EXPECT_TRUE(bundleATracked) << "GetBundles() should include the test bundle";
+
+  bundleTracker.Close();
+  bundles = bundleTracker.GetBundles();
+  EXPECT_EQ(0, bundles.size())
+    << "GetBundles() should return an empty vector after Close()";
+}
+
+TEST_F(BundleTrackerMethodTest, GetObjectWorks)
+{
+  // Given an open BundleTracker not tracking custom objects
+  BundleTracker<> bundleTracker(context, all_states);
+  ASSERT_NO_THROW(bundleTracker.Open()) << "BundleTracker failed to start";
+
+  // When a bundle is tracked
+  Bundle bundleA = cppmicroservices::testing::InstallLib(
+    framework.GetBundleContext(), "TestBundleA");
+  auto bundles = bundleTracker.GetBundles();
+  bool bundleATracked =
+    std::find(bundles.begin(), bundles.end(), bundleA) != bundles.end();
+  ASSERT_TRUE(bundleATracked) << "A bundle that should be tracked is untracked";
+
+  // Then GetObject should return the bundle
+  EXPECT_EQ(bundleA, bundleTracker.GetObject(bundleA))
+    << "GetObject for a tracked bundle should return the bundle when no custom "
+       "objects were used";
+
+  // When the bundle is untracked, then GetObject should return nullopt
+  bundleTracker.Remove(bundleA);
+  EXPECT_EQ(std::nullopt, bundleTracker.GetObject(bundleA))
+    << "GetObject for an untracked, valid bundle should return nullopt";
+
+  // When the bundle is invalid, then GetObject should return nullopt
+  Bundle invalidBundle = Bundle();
+  ASSERT_NO_THROW(bundleTracker.GetObject(invalidBundle))
+    << "GetObject threw an error for an invalid bundle";
+  EXPECT_EQ(std::nullopt, bundleTracker.GetObject(invalidBundle))
+    << "GetObject for an untracked, invalid bundle should return nullopt";
+
+  bundleTracker.Close();
+}
+
+TEST_F(BundleTrackerMethodTest, GetTrackedWorks)
```

Comment:
minor; rename testpoint name - remove "Works" to be more consistent with other test point names

Reply:
Done.